File: Mario_Sokoban_Version_Python/v1.0/jeu.py
#coding:utf-8

# -------------- Mario Sokoban v1.0 -----------------------------------


"""
	#Monday july 12 2021 00:31 debut de developpement
	jeu.py
-----------------
	Par Mohamed kamara 
	
	Role : Gere une partie de complete

	Modification a faire : faire accessible la variable positionJoueur qui ne peut pas etre modifier par les fonctions jouer et deplacerJoueur

"""
from data import *
import fichiers
import pygame

#----------------------------------------------------------------------------------------
def jouer(window_surface):
	pygame.init()	
	global grille
#VARIABLE
	#Personnage
		#MARIO
	mario = {"haut" : 0, "bas" : 0, "gauche" : 0, "droite" : 0}
	
	mario["haut"] = pygame.image.load("src_img/mario_haut.gif")
	mario["haut"].convert()
	
	mario["bas"] = pygame.image.load("src_img/mario_bas.gif")
	mario["bas"].convert()
	
	mario["gauche"] = pygame.image.load("src_img/mario_gauche.gif")
	mario["gauche"].convert()
	
	mario["droite"] = pygame.image.load("src_img/mario_droite.gif")
	mario["droite"].convert()



	#Chargement des sprites
	mur = pygame.image.load("src_img/mur.jpg")
	mur.convert()
	
	caisse = pygame.image.load("src_img/caisse.jpg")
	caisse.convert()
	
	caisse_ok = pygame.image.load("src_img/caisse_ok.jpg")
	caisse_ok.convert()
	
	objectif = pygame.image.load("src_img/objectif.png")
	objectif.convert()

	#Autre
	position = [0, 0]
	global positionJoueur
	objectifRestant = False

	joueurActuel = mario["bas"]


	#Chargement du niveau
	if fichiers.chargerNiveau() == False:
		exit("Impossible de charger le niveau\n No such 'level.lvl'")

	#Recherche de la position de mario
	i = 0 
	while i < nb_bloc_width:
		j = 0
		while j < nb_bloc_height:
			if grille[i][j] == items["MARIO"]:
				positionJoueur = [i, j]
				grille[i][j] = items["VIDE"]
			j += 1
		i += 1	

#BOUCLE PRINCIPALE
	jeu_start = True
	while jeu_start:
	#Condition et capture des evenements
		for event in pygame.event.get():
			if event.type == pygame.QUIT:
				jeu_start = False
			elif event.type == pygame.KEYDOWN:
				if event.key == pygame.K_UP:
					joueurActuel = mario["haut"]
					deplacerJoueur(grille, direction["HAUT"])
				elif event.key == pygame.K_DOWN:
					joueurActuel = mario["bas"]
					deplacerJoueur(grille, direction["BAS"])
				elif event.key == pygame.K_LEFT:
					joueurActuel = mario["gauche"]
					deplacerJoueur(grille, direction["GAUCHE"])
				elif event.key == pygame.K_RIGHT:
					joueurActuel = mario["droite"]
					deplacerJoueur(grille, direction["DROITE"])

	#Blitage des élements
		window_surface.fill(white_color)

		objectifRestant = False

		i = 0
		while i < nb_bloc_width:
			j = 0
			while j < nb_bloc_height:

				position = [i* bloc_size, j*bloc_size]

				if grille[i][j] == items["CAISSE"]:
					window_surface.blit(caisse, position)
				elif grille[i][j] == items["CAISSE_OK"]:
					window_surface.blit(caisse_ok, position)
				elif grille[i][j] == items["MUR"]:
					window_surface.blit(mur, position)
				elif grille[i][j] == items["OBJECTIF"]:
					window_surface.blit(objectif, position)
					objectifRestant = True
				j += 1
			i += 1

		#check win
		if objectifRestant == False:
			jeu_start = False



		#Blitage du joueur a l'ecran
		position = [(positionJoueur[0]*bloc_size) , (positionJoueur[1]*bloc_size)]
		window_surface.blit(joueurActuel, position)


		#Mise a jour de l'ecran
		pygame.display.flip()


#---------------------------------------------------------------------------------------------------------------------------------
def deplacerJoueur(grille, direction_sens):
	"""
		Fonction deplacerJoueur depalace le joueur d'une case a l'autre,
		prend en parametre la grille c'est a dire la carte du jeu et la direction dans laquelle deplacée le joueur
		Mais aussi rend accessible la variable positionJoueur pour modifier sa position 
	"""
	global positionJoueur

	if direction_sens == direction["HAUT"]:
		if positionJoueur[1] - 1 < 0: #Si le joueur depasse l'ecran, on arrete
			return False
		elif grille[positionJoueur[0]][positionJoueur[1]-1] == items["MUR"]: #s'il y a un mur au dessus du joueur on arrete
			return False
		#Si on veut pousser une caisse, il faut verifier qu'il n'y a pas de caisse apres ou le mur ou la limite du jeu
		elif (grille[positionJoueur[0]][positionJoueur[1]-1] == items["CAISSE"] or grille[positionJoueur[0]][positionJoueur[1]-1] == items["CAISSE_OK"]) and ((positionJoueur[1] - 2 < 0) or grille[positionJoueur[0]][positionJoueur[1]-2] == items["MUR"] or grille[positionJoueur[0]][positionJoueur[1]-2] == items["CAISSE"] or grille[positionJoueur[0]][positionJoueur[1]-2] == items["CAISSE_OK"]) :
			return False
		else:
			deplacerCaisse(positionJoueur, direction["HAUT"])
			positionJoueur = [positionJoueur[0], positionJoueur[1] - 1] #Modifie l'ordonnée du joueur pour le faire monter d'une case
			return True

	elif direction_sens == direction["BAS"]:
		if positionJoueur[1] + 1 >= nb_bloc_width: #Si le joueur depasse l'ecran, on arrete
			return False
		elif grille[positionJoueur[0]][positionJoueur[1]+1] == items["MUR"]: #s'il y a un mur au dessus du joueur on arrete
			return False
		elif (grille[positionJoueur[0]][positionJoueur[1]+1] == items["CAISSE"] or grille[positionJoueur[0]][positionJoueur[1]+1] == items["CAISSE_OK"] ) and ((positionJoueur[1] + 2 < 0) or grille[positionJoueur[0]][positionJoueur[1]+2] == items["MUR"] or grille[positionJoueur[0]][positionJoueur[1]+2] == items["CAISSE"] or grille[positionJoueur[0]][positionJoueur[1]+2] == items["CAISSE_OK"]) :
			return False	
		else:
			deplacerCaisse(positionJoueur, direction["BAS"])
			positionJoueur = [positionJoueur[0], positionJoueur[1] + 1] #Modifie l'ordonnée du joueur pour le faire monter d'une case
			return True

	if direction_sens == direction["GAUCHE"]:
		if positionJoueur[0] - 1 < 0: #Si le joueur depasse l'ecran, on arrete
			return False
		elif grille[positionJoueur[0]-1][positionJoueur[1]] == items["MUR"]: #s'il y a un mur au dessus du joueur on arrete
			return False
		elif (grille[positionJoueur[0]-1][positionJoueur[1]] == items["CAISSE"] or grille[positionJoueur[0]-1][positionJoueur[1]] == items["CAISSE_OK"]) and ((positionJoueur[0] - 2 < 0) or grille[positionJoueur[0]-2][positionJoueur[1]] == items["MUR"] or grille[positionJoueur[0]-2][positionJoueur[1]] == items["CAISSE"] or grille[positionJoueur[0]-2][positionJoueur[1]] == items["CAISSE_OK"]) :
			return False
		else:
			deplacerCaisse(positionJoueur, direction["GAUCHE"])
			positionJoueur = [positionJoueur[0] - 1, positionJoueur[1]] #Modifie l'ordonnée du joueur pour le faire monter d'une case
			return True

	if direction_sens == direction["DROITE"]:
		if positionJoueur[0] + 1 >= nb_bloc_height: #Si le joueur depasse l'ecran, on arrete
			return False
		elif grille[positionJoueur[0]+1][positionJoueur[1]] == items["MUR"]: #s'il y a un mur au dessus du joueur on arrete
			return False
		elif (grille[positionJoueur[0]+1][positionJoueur[1]] == items["CAISSE"] or grille[positionJoueur[0]+1][positionJoueur[1]] == items["CAISSE_OK"]) and ((positionJoueur[0] + 2 < 0) or grille[positionJoueur[0]+2][positionJoueur[1]] == items["MUR"] or grille[positionJoueur[0]+2][positionJoueur[1]] == items["CAISSE"] or grille[positionJoueur[0]+2][positionJoueur[1]] == items["CAISSE_OK"]) :
			return False
		else:
			deplacerCaisse(positionJoueur, direction["DROITE"])
			positionJoueur = [positionJoueur[0] + 1, positionJoueur[1]] #Modifie l'ordonnée du joueur pour le faire monter d'une case
			return True
 



#-------------------------------------------------------------------------------------------------------------------
def deplacerCaisse(position, direction_sens):
	"""
		fonction deplacerCaisse, prend en parametre la position du joueur qu'elle ne modifie pas
		et la direction dans laquelle deplacée la caisse
		Et rend accessible la grille (la carte) qu'elle modifiera
	"""
	global grille

	if direction_sens == direction["HAUT"]:
		if grille[position[0]][position[1]-1] == items["CAISSE"] or grille[position[0]][position[1]-1] == items["CAISSE_OK"]:
			if grille[position[0]][position[1]-2] == items["OBJECTIF"]:
				grille[position[0]][position[1]-2] = items["CAISSE_OK"]
			else:
				grille[position[0]][position[1]-2] = items["CAISSE"]

			if grille[position[0]][position[1]-1] == items["CAISSE_OK"]:
				grille[position[0]][position[1]-1] = items["OBJECTIF"]
			else:
				grille[position[0]][position[1]-1] = items["VIDE"]

	elif direction_sens == direction["BAS"]:
		if grille[position[0]][position[1]+1] == items["CAISSE"] or grille[position[0]][position[1]+1] == items["CAISSE_OK"]:
			if grille[position[0]][position[1]+2] == items["OBJECTIF"]:
				grille[position[0]][position[1]+2] = items["CAISSE_OK"]
			else:
				grille[position[0]][position[1]+2] = items["CAISSE"]

			if grille[position[0]][position[1]+1] == items["CAISSE_OK"]:
				grille[position[0]][position[1]+1] = items["OBJECTIF"]
			else:
				grille[position[0]][position[1]+1] = items["VIDE"]

	elif direction_sens == direction["GAUCHE"] :
		if grille[position[0]-1][position[1]] == items["CAISSE"] or grille[position[0]-1][position[1]] == items["CAISSE_OK"]:
			if grille[position[0]-2][position[1]] == items["OBJECTIF"]:
				grille[position[0]-2][position[1]] = items["CAISSE_OK"]
			else:
				grille[position[0]-2][position[1]] = items["CAISSE"]

			if grille[position[0]-1][position[1]] == items["CAISSE_OK"]:
				grille[position[0]-1][position[1]] = items["OBJECTIF"]
			else:
				grille[position[0]-1][position[1]] = items["VIDE"]
				
	elif direction_sens == direction["DROITE"]:
		if grille[position[0]+1][position[1]] == items["CAISSE"] or grille[position[0]+1][position[1]] == items["CAISSE_OK"]:
			if grille[position[0]+2][position[1]] == items["OBJECTIF"]:
				grille[position[0]+2][position[1]] = items["CAISSE_OK"]
			else:
				grille[position[0]+2][position[1]] = items["CAISSE"]

			if grille[position[0]+1][position[1]] == items["CAISSE_OK"]:
				grille[position[0]+1][position[1]] = items["OBJECTIF"]
			else:
				grille[position[0]+1][position[1]] = items["VIDE"]
				

File: Mario_Sokoban_Version_C/jeu.c
/*
	jeu.c

	----------------------

	Par Mohamedkamara, pour un exo TP

	Rôle : gere une partie de jeu complete
*/
#include <SDL/SDL.h>
#include <SDL/SDL_image.h>
#include <stdlib.h>
#include <stdio.h>

#include "jeu.h"
#include "fichiers.h"

/*-----------------------------------------------------------------------------------------------------------------------------------*/
void jouer(SDL_Surface* ecran)
{
//Declaration des variables
	SDL_Surface *mario[4] = {NULL}; //4 surfaces pour 4 directions de mario
	SDL_Surface *mur = NULL, *caisse = NULL, *caisse_ok = NULL, *objectif = NULL, *instructions = NULL, *marioActuel = NULL, *bravo = NULL;//marioActuel pointe vers une case du tableu de mario
	SDL_Rect position, positionJoueur, pos_instructions, pos_bravo;
	SDL_Event event;
	
	FILE *fichier = NULL;
	fichier = fopen("stdout.txt", "a+");
	if(fichier == NULL)
	{
		fprintf(stderr, "Impossible d'ouvrir le fichier 'stdout.txt'\n");
		exit(EXIT_FAILURE);
	}
	fprintf(fichier, "Time: %s --------- Date: %s ---------- File: %s --------- Line: %d\n", __TIME__, __DATE__, __FILE__, __LINE__);
	int continuer = 1, objectifsRestants = 0, i = 0, j= 0, continu = 1;
	int carte[NB_BLOC_LARGEUR][NB_BLOC_HAUTEUR] = {0}, niveau = 0; 
	pos_instructions.x = 0; pos_instructions.y = 0;

//Chargement des sprites(image, personnage...)
	mur = IMG_Load("src_img/mur.jpg");
	caisse = IMG_Load("src_img/caisse.jpg");
	caisse_ok = IMG_Load("src_img/caisse_ok.jpg");
	objectif = IMG_Load("src_img/objectif.png");
	instructions = IMG_Load("src_img/instructions_jeu.jpg");
	bravo = IMG_Load("src_img/bravo.jpg");
	mario[BAS] = IMG_Load("src_img/mario_bas.gif");
	mario[HAUT] = IMG_Load("src_img/mario_haut.gif");
	mario[GAUCHE] = IMG_Load("src_img/mario_gauche.gif");
	mario[DROITE] = IMG_Load("src_img/mario_droite.gif");

//Orientation initiale de mario
	marioActuel = mario[BAS];

//Chargement du niveau
	if (!chargerNiveau(carte, niveau))
	{
		fprintf(fichier, "Time: %s --------- Date: %s ---------- File: %s --------- Line: %d\n", __TIME__, __DATE__, __FILE__, __LINE__);
		fprintf(fichier, "Impossible de charger le niveau : cause possible le fichier 'niveaux.lvl' n'existe pas.\n");
		exit(EXIT_FAILURE);
	}

//Recherche de la position de mario sur la carte
	for (i = 0; i < NB_BLOC_LARGEUR; i++)
	{
		for (j = 0; j < NB_BLOC_HAUTEUR; j++)
		{
			if (carte[i][j] == MARIO)//Si mario se trouve a cette position
			{
				positionJoueur.x = i;
				positionJoueur.y = j;
				carte[i][j] = VIDE;
			}
		}
	}

//Activation de la repetiton des touches
	SDL_EnableKeyRepeat(100, 100);
//Affichage des instructions
	while(continuer)
	{
		SDL_WaitEvent(&event);
		switch(event.type)
		{
			case SDL_KEYDOWN:
				continuer = 0;
		}
		SDL_FillRect(ecran, NULL, SDL_MapRGB(ecran->format, 255, 255, 255));
		SDL_BlitSurface(instructions, NULL, ecran, &pos_instructions);
		SDL_Flip(ecran);
	}

//Boucle principale
	continuer = 1;
	while(continuer)
	{
		SDL_WaitEvent(&event);
		switch(event.type)
		{
			case SDL_QUIT :
				continuer = 0;
				break;
			case SDL_KEYDOWN:
				switch(event.key.keysym.sym)
				{
					case SDLK_ESCAPE:
						continuer = 0;
						break;
					case SDLK_r:
						if (!chargerNiveau(carte, niveau))
						{
							fprintf(fichier, "Time: %s --------- Date: %s ---------- File: %s --------- Line: %d\n", __TIME__, __DATE__, __FILE__, __LINE__);
							fprintf(fichier, "Impossible de charger le niveau : cause possible le fichier 'niveaux.lvl' n'existe pas.\n");
							exit(EXIT_FAILURE);
						}
						for (i = 0; i < NB_BLOC_LARGEUR; i++)
						{
							for (j = 0; j < NB_BLOC_HAUTEUR; j++)
							{
								if (carte[i][j] == MARIO)//Si mario se trouve a cette position
								{
									positionJoueur.x = i;
									positionJoueur.y = j;
									carte[i][j] = VIDE;
								}
							}
						}
						marioActuel = mario[BAS];
						break;
					case SDLK_UP:
						marioActuel = mario[HAUT];
						deplacerJoueur(carte, &positionJoueur, HAUT);
						break;
					case SDLK_DOWN:
						marioActuel = mario[BAS];
						deplacerJoueur(carte, &positionJoueur, BAS);
						break;
					case SDLK_RIGHT:
						marioActuel = mario[DROITE];
						deplacerJoueur(carte, &positionJoueur, DROITE);
						break;
					case SDLK_LEFT:
						marioActuel = mario[GAUCHE];
						deplacerJoueur(carte, &positionJoueur, GAUCHE);
						break;
				}
				break;
		}
		//Effacement de l'ecrant en coloriant en blanc
		SDL_FillRect(ecran, NULL, SDL_MapRGB(ecran->format, 255, 255, 255));

		//Placement des objets a l'ecran
		objectifsRestants = 0;
		
		for (i = 0; i < NB_BLOC_LARGEUR; i++)
		{
			for(j = 0; j < NB_BLOC_HAUTEUR; j++)
			{
				position.x = i * TAILLE_BLOC;
				position.y = j * TAILLE_BLOC;

				switch(carte[i][j])
				{
					case MUR:
						SDL_BlitSurface(mur, NULL, ecran, &position);
						break;
					case CAISSE:
						SDL_BlitSurface(caisse, NULL, ecran, &position);
						break;
					case CAISSE_OK:
						SDL_BlitSurface(caisse_ok, NULL, ecran, &position);
						break;
					case OBJECTIF:
						SDL_BlitSurface(objectif, NULL, ecran, &position);
						objectifsRestants = 1;
						break;
				}

			}
		}

		//Si on a trouvé aucun objectif sur la carte sa veut dire qu"on a gagné
		if(!objectifsRestants)
			{
			//Chargement du niveau
				niveau++;
				if (!chargerNiveau(carte, niveau))
				{
					fprintf(fichier, "Time: %s --------- Date: %s ---------- File: %s --------- Line: %d\n", __TIME__, __DATE__, __FILE__, __LINE__);
					fprintf(fichier, "Impossible de charger le niveau : cause possible le fichier 'niveaux.lvl' n'existe pas.\n");
					exit(EXIT_FAILURE);
				}
				for (i = 0; i < NB_BLOC_LARGEUR; i++)
				{
					for (j = 0; j < NB_BLOC_HAUTEUR; j++)
					{
						if (carte[i][j] == MARIO)//Si mario se trouve a cette position
						{
							positionJoueur.x = i;
							positionJoueur.y = j;
							carte[i][j] = VIDE;
						}
					}
				}
				marioActuel = mario[BAS];
				//Affichage de bravo
					pos_bravo.x = (ecran->w / 2) - (bravo->w / 2);
					pos_bravo.y = (ecran->h / 2) - (bravo->h / 2);
				while(continu)
				{
					SDL_WaitEvent(&event);
					switch(event.type)
					{
						case SDL_KEYDOWN:
							continu = 0;
					}
					//SDL_FillRect(ecran, NULL, SDL_MapRGB(ecran->format, 255, 255, 255));
					SDL_BlitSurface(bravo, NULL, ecran, &pos_bravo);
					SDL_Flip(ecran);
				}
				continu = 1;
			}

		//On place le joueur a la bonne position
		position.x = positionJoueur.x * TAILLE_BLOC;
		position.y = positionJoueur.y * TAILLE_BLOC;
		SDL_BlitSurface(marioActuel, NULL, ecran, &position);

		//Mise a jour de l'ecran 
		SDL_Flip(ecran);
	}

//Desactivation de la repetiton des touches
	SDL_EnableKeyRepeat(0, 0);

//Liberation de surface
	SDL_FreeSurface(mur);
	SDL_FreeSurface(caisse);
	SDL_FreeSurface(caisse_ok);
	SDL_FreeSurface(instructions);
	SDL_FreeSurface(objectif);
	for (i = 0; i < 4; ++i)
		SDL_FreeSurface(mario[i]);
	fclose(fichier);
}

/*-----------------------------------------------------------------------------------------------------------------------------------*/
void deplacerJoueur(int carte[][NB_BLOC_HAUTEUR], SDL_Rect *pos, int direction)
{
	switch(direction)
	{
		case HAUT:
		//Verification si le joueur a le droit de se deplacer ou de deplacer une caisse
			if(pos->y - 1 < 0)//Si le joueur se trouve au bord de la fenetre
				break;

			if (carte[pos->x][pos->y - 1] == MUR)//S'il y a un mur on arrete
				break;

			//Si on veut pousser une caisse il faut verifier qu'il y a pas un mur derriere ou une autre caisse ou le bord de la fenetre
			if((carte[pos->x][pos->y - 1] == CAISSE_OK || carte[pos->x][pos->y - 1] == CAISSE) && (pos->y - 2 < 0 || carte[pos->x][pos->y - 2] == MUR || carte[pos->x][pos->y - 2] == CAISSE || carte[pos->x][pos->y - 2] == CAISSE_OK))
				break;

			//SI on arrive là c'est qu'on peut deplacer le joueur
			//On verifie d'abord s'il y a pas de caisse a deplacer
			deplacerCaisse(&carte[pos->x][pos->y - 1], &carte[pos->x][pos->y - 2]);

			//On peut enfin faire monter le joueur Ouff !!
			pos->y--;
			break;

		case BAS:
			//Verification si le joueur a le droit de se deplacer ou de deplacer une caisse
			if(pos->y + 1 >= NB_BLOC_HAUTEUR)//Si le joueur se trouve au bord de la fenetre
				break;

			if (carte[pos->x][pos->y + 1] == MUR)//S'il y a un mur on arrete
				break;

			//Si on veut pousser une caisse il faut verifier qu'il y a pas un mur derriere ou une autre caisse ou le bord de la fenetre
			if((carte[pos->x][pos->y + 1] == CAISSE_OK || carte[pos->x][pos->y + 1] == CAISSE) && (pos->y + 2 >= NB_BLOC_HAUTEUR || carte[pos->x][pos->y + 2] == MUR || carte[pos->x][pos->y + 2] == CAISSE || carte[pos->x][pos->y + 2] == CAISSE_OK))
				break;

			//SI on arrive là c'est qu'on peut deplacer le joueur
			//On verifie d'abord s'il y a pas de caisse a deplacer
			deplacerCaisse(&carte[pos->x][pos->y + 1], &carte[pos->x][pos->y + 2]);

			//On peut enfin faire monter le joueur Ouff !!
			pos->y++;
			break;
		case GAUCHE:
		//Verification si le joueur a le droit de se deplacer ou de deplacer une caisse
			if(pos->x - 1 < 0)//Si le joueur se trouve au bord de la fenetre
				break;

			if (carte[pos->x - 1][pos->y] == MUR)//S'il y a un mur on arrete
				break;

			//Si on veut pousser une caisse il faut verifier qu'il y a pas un mur derriere ou une autre caisse ou le bord de la fenetre
			if((carte[pos->x- 1][pos->y] == CAISSE_OK || carte[pos->x - 1][pos->y] == CAISSE) && (pos->x - 2 < 0 || carte[pos->x - 2][pos->y] == MUR || carte[pos->x - 2][pos->y] == CAISSE || carte[pos->x - 2][pos->y] == CAISSE_OK))
				break;

			//SI on arrive là c'est qu'on peut deplacer le joueur
			//On verifie d'abord s'il y a pas de caisse a deplacer
			deplacerCaisse(&carte[pos->x - 1][pos->y], &carte[pos->x - 2][pos->y]);

			//On peut enfin faire monter le joueur Ouff !!
			pos->x--;
			break;
		case DROITE:
		//Verification si le joueur a le droit de se deplacer ou de deplacer une caisse
			if(pos->x + 1 >= NB_BLOC_LARGEUR)//Si le joueur se trouve au bord de la fenetre
				break;

			if (carte[pos->x + 1][pos->y] == MUR)//S'il y a un mur on arrete
				break;

			//Si on veut pousser une caisse il faut verifier qu'il y a pas un mur derriere ou une autre caisse ou le bord de la fenetre
			if((carte[pos->x + 1][pos->y] == CAISSE_OK || carte[pos->x + 1][pos->y] == CAISSE) && (pos->x + 2 >= NB_BLOC_LARGEUR|| carte[pos->x + 2][pos->y] == MUR || carte[pos->x + 2][pos->y] == CAISSE || carte[pos->x + 2][pos->y] == CAISSE_OK))
				break;

			//SI on arrive là c'est qu'on peut deplacer le joueur
			//On verifie d'abord s'il y a pas de caisse a deplacer
			deplacerCaisse(&carte[pos->x + 1][pos->y], &carte[pos->x + 2][pos->y]);

			//On peut enfin faire monter le joueur Ouff !!
			pos->x++;
			break;
	}
}

/*----------------------------------------------------------------------------------------------------------------------------------*/
void deplacerCaisse(int *premierCase, int *secondeCase)
{
	if (*premierCase == CAISSE || *premierCase == CAISSE_OK)
	{
		if(*secondeCase == OBJECTIF)
			*secondeCase = CAISSE_OK;
		else
			*secondeCase = CAISSE;

		if(*premierCase == CAISSE_OK) 
			*premierCase = OBJECTIF;
		else 
			*premierCase = VIDE;
	}
}